iroh-net/src/magicsock/node_map/node_state.rs

```diff
@@ -128,6 +128,9 @@ pub(super) struct NodeState {
     last_call_me_maybe: Option<Instant>,
     /// The type of connection we have to the node, either direct, relay, mixed, or none.
     conn_type: Watchable<ConnectionType>,
+    /// How we learned about this node. We can have learned about this node from multiple stores.
+    /// Sources later in the list were sources we learned from more recently.
+    sources: Vec<(super::Source, Instant)>,
```

Comment:
I'm a bit worried this can only grow, also from checking the output in the console I think I might want is the source of the particular addresses rather than the full node, not sure what are other's thoughts

Reply:
Having sources on a per-address basis is interesting, and as we talk about beefing up the node map / writing a node address book, this could be something we want to look into.

Right now, the only thing users have requested was a way to determine how a node was discovered, not about particular addresses.

I agree it's concerning to have this only grow. Maybe we only keep the last X number?